src/NUnitFramework/testdata/TimeoutFixture.cs

```diff
@@ -108,4 +108,50 @@ public void TestTimeOutTestCase(int delay)
             Thread.Sleep(delay);
         }
     }
+
+    [TestFixture]
+    public class TimeoutWithSetupAndOutputAfterTimeoutFixture
+    {
+        [SetUp]
+        public void Setup()
+        {
+            TestContext.WriteLine("setup");
+        }
+
+        [Test, Timeout(600)]
+        public void Test2()
+        {
+            TestContext.WriteLine("method output before pause");
+            Thread.Sleep(1_000);
+            TestContext.WriteLine("method output after pause");
+            Assert.That(1, Is.EqualTo(0));
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+        }
+    }
+
+    [TestFixture]
+    public class TimeoutWithSetupAndOutputFixture
+    {
+        [SetUp]
+        public void Setup()
+        {
+            TestContext.WriteLine("setup");
+        }
+
+        [Test, Timeout(2_000)]
+        public void Test2()
+        {
+            TestContext.WriteLine("method output");
+            Assert.That(1, Is.EqualTo(0));
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
```

Comment:
The #4598 issue talks about `TearDown` output being lost. We need a test with that in.

Reply:
Good point! I had changed the test from having a `TearDown` to a `SetUp` during development and missed re-adding a separate `TearDown` case. I'll add one.

I think the originating issue sought more to keep output from the test method itself in the presence of a `TearDown` rather than output from the `TearDown` itself but that will be a good test for me to add here to. One thing to note here though is that I think this was a NET6-specific bug and that presently `TearDown` is presently only run for timedout test cases on netfx (https://github.com/nunit/nunit/issues/2397). I verified the behaviour described there and it's consistent as far back as 3.12

Reply:
Wait, sorry. @manfred-brands I think this does cover the example in the issue as it talks about test output from the test rather than the `TearDown`, but I'll still take another look and see if there's any edge cases I could add